packages/gatsby/src/utils/page-data.ts

```diff
@@ -196,7 +196,8 @@ export async function flush(): Promise<void> {
 
         if (hasFlag(query.dirty, FLAG_DIRTY_NEW_PAGE)) {
           // query results are not written yet
-          return cb(null, true)
+          process.nextTick(() => cb(null, true))
+          return
```

Comment:
Without this fastq will jump to the next task in the same tick and if you have too many DSG tasks - it will overflow the call stack.